File: 프로그래머스/lv2/12941. 최솟값 만들기/최솟값 만들기.js
function solution(A,B){
    let result = 0;
    if(Math.max(...A) * Math.min(...B) > Math.max(...B) * Math.min(...A)) {
        B = B.sort((a,b) => a - b);
        A = A.sort((a,b) => b - a); 
    } else {
        A = A.sort((a,b) => a - b);
        B = B.sort((a,b) => b - a); 
    }
   
    for(let i = 0; i < A.length; i++) {
        result += (A[i] * B[i])
    }
    return result
}

File: 프로그래머스/lv2/12941. 최솟값 만들기/최솟값 만들기.java
import java.util.*;
class Solution{
    public int solution(int []A, int []B){
        int answer = 0;
        Arrays.sort(A);
        Integer[] blist = Arrays.stream(B).boxed().toArray(Integer[]::new);
        Arrays.sort(blist,Collections.reverseOrder());
        for(int i = 0; i < A.length; i++) {
            answer += (A[i] * blist[i]);
        }
        return answer;
    }
}